test/common/upstream/test_local_address_selector.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "envoy/upstream/upstream.h"
+
+namespace Envoy {
+namespace Upstream {
+
+class TestUpstreamLocalAddressSelector : public UpstreamLocalAddressSelector {
+public:
+  TestUpstreamLocalAddressSelector(
+      std::vector<::Envoy::Upstream::UpstreamLocalAddress> upstream_local_addresses,
+      std::shared_ptr<size_t> num_calls, bool return_empty_source_address)
```

Comment:
I didn't see anywhere set the `return_empty_source_address` as false

Reply:
It's set in the factory on line 40, but I can set a default here too.